substrate/dex/pallet/src/lib.rs

```diff
@@ -154,11 +154,34 @@ pub mod pallet {
   #[pallet::storage]
   pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId, PoolInfo<Coin>, OptionQuery>;
 
+  #[pallet::storage]
+  #[pallet::getter(fn last_quote_for_block)]
+  pub type LastQuoteForBlock<T: Config> =
+    StorageDoubleMap<_, Identity, BlockNumberFor<T>, Identity, Coin, [u8; 8], ValueQuery>;
+
+  /// Moving window of oracle prices.
+  #[pallet::storage]
+  #[pallet::getter(fn oracle_prices)]
+  pub type OraclePrices<T: Config> = StorageMap<_, Identity, [u8; 8], u16, OptionQuery>;
```

Comment:
I'd document the [u8; 8] is the amount's big endian bytes, and u16 is the amount of inclusions in this multi-set.

We also should have an on-genesis assert the window size wasn't configured to be >= 2**16 (which would risk an overflow here).

Reply:
You're also missing the Coin. You need a DoubleMap with Coin, [u8; 8] or a Map with (Coin, [u8;8]).